spec.html

```diff
@@ -709,32 +689,171 @@
+        <emu-alg>
+          1. Return Completion{[[Type]]: ~throw~, [[Value]]: a newly created *TypeError* object, [[Target]]: ~empty~}.
+        </emu-alg>
+      </emu-clause>
+
+      <!-- es6num="6.2.2.4" -->
+      <emu-clause id="sec-returnifabrupt" aoid="ReturnIfAbrupt">
+        <h1>ReturnIfAbrupt</h1>
+        <p>Algorithms steps that say or are otherwise equivalent to:</p>
+        <emu-alg>
+          1. ReturnIfAbrupt(_argument_).
+        </emu-alg>
+        <p>mean the same thing as:</p>
+        <emu-alg>
+          1. If _argument_ is an abrupt completion, return _argument_.
+          1. Else if _argument_ is a Completion Record, let _argument_ be _argument_.[[Value]].
+        </emu-alg>
+        <p>Algorithms steps that say or are otherwise equivalent to:</p>
+        <emu-alg>
+          1. ReturnIfAbrupt(AbstractOperation()).
+        </emu-alg>
+        <p>mean the same thing as:</p>
+        <emu-alg>
+          1. Let _hygienicTemp_ be AbstractOperation().
+          1. If _hygienicTemp_ is an abrupt completion, return _hygienicTemp_.
+          1. Else if _hygienicTemp_ is a Completion Record, let _hygienicTemp_ be _hygienicTemp_.[[Value]].
+        </emu-alg>
+        <p>Where _hygienicTemp_ is ephemeral and visible only in the steps pertaining to ReturnIfAbrupt.</p>
+        <p>Algorithms steps that say or are otherwise equivalent to:</p>
+        <emu-alg>
+          1. Let _result_ be AbstractOperation(ReturnIfAbrupt(_argument_)).
+        </emu-alg>
+        <p>mean the same thing as:</p>
+        <emu-alg>
+          1. If _argument_ is an abrupt completion, return _argument_.
+          1. If _argument_ is a Completion Record, let _argument_ be _argument_.[[Value]].
+          1. Let _result_ be AbstractOperation(_argument_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-returnifabrupt-shorthands">
+        <h1>ReturnIfAbrupt Shorthands</h1>
+        <p>Abstract operations referenced using the functional application style and the method application style that are prefixed by `?` indicate that ReturnIfAbrupt should be applied to the resulting Completion Record. For example, the algorithm:</p>
+        <emu-alg>
+          1. ? operationName().
+        </emu-alg>
+        <p>is equivalent to the following algorithm:</p>
+        <emu-alg>
+          1. ReturnIfAbrupt(operationName()).
+        </emu-alg>
+        <p>Similarly, for method application style:</p>
+        <emu-alg>
+          1. ? _someValue_.operationName().
+        </emu-alg>
+        <p>is equivalent to the following algorithm:</p>
+        <emu-alg>
+          1. ReturnIfAbrupt(_someValue_.operationName()).
+        </emu-alg>
+        <p>Similarly, prefix `!` is used to indicate that an abstract operation will never return an abrupt completion and that the resulting Completion Record's value field should be used in place of the return value of the operation. For example, the algorithm:</p>
```

Comment:
I like this suggestion.

_this invocation_ or _the following invocation_. Also, trying to keep the most from the original: "... is used to indicate __the following__ abstract operation_ ..."